presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java

```diff
@@ -250,6 +251,14 @@ public QueryInfo getFullQueryInfo()
                 .orElseGet(() -> stateMachine.updateQueryInfo(Optional.empty()));
     }
 
+    @Override
+    public QueryState getState()
+    {
+        return tryGetQueryExecution()
+                .map(queryExecution -> queryExecution.getQueryInfo().getState())
```

Comment:
obtaining query info is even more expensive than basic stats

Reply:
Ugh.  `stateMachine.getState()`?

Reply:
Yes. That one is cheap

Reply:
Thanks for catching.